### PR TITLE
Angular 15 fixes

### PR DESCRIPTION
**Pull request by maxokorokov**
More minor fixes for Angular 15

### CODE REVIEW

File: demo/src/main.ts [hunk 1]
```diff
@@ -13,6 +12,6 @@ bootstrapApplication(AppComponent, {
 			provide: LocationStrategy,
 			useClass: HashLocationStrategy,
 		},
-		importProvidersFrom([HttpClientModule]),
+		provideHttpClient(),
```
Comment (jnizet) [inline demo/src/main.ts:15]: nit: we could use `provideHttpClient(withNoXsrfProtection())`

File: misc/api-doc.ts [hunk 2]
```diff
@@ -63,6 +83,32 @@ function isInternalMember(member) {
...
+		isIdentifier(decorator.expression.expression) &&
+		types.includes(decorator.expression.expression.text)
+	);
+}
+
+function getDecoratorOfType(node: Node, decoratorType: string): Decorator | null {
+	const decorators = canHaveDecorators(node) ? getDecorators(node) || [] : [];
+
+	for (const decorator of decorators) {
+		if (isCallExpression(decorator.expression) && isIdentifier(decorator.expression.expression)) {
```
Comment (jnizet) [inline misc/api-doc.ts:98]: Couldn't we use `isDecoratorOfType` here rather than duplicating the logic?

File: misc/api-doc.ts [hunk 3]
```diff
@@ -130,22 +150,21 @@ class APIDocVisitor {
...
+			} else if (isInterfaceDeclaration(statement)) {
 				return directivesSoFar.concat(this.visitInterfaceDeclaration(fileName, statement));
 			}
 
 			return directivesSoFar;
 		}, []);
 	}
 
-	visitInterfaceDeclaration(fileName, interfaceDeclaration) {
+	visitInterfaceDeclaration(fileName, interfaceDeclaration: InterfaceDeclaration) {
```
Comment (jnizet) [inline misc/api-doc.ts:163]: fileName: string

File: misc/api-doc.ts [hunk 4]
```diff
@@ -314,80 +362,43 @@ class APIDocVisitor {
...
-			defaultValue: property.initializer ? this.stringifyDefaultValue(property.initializer) : undefined,
-			type: this.visitType(property),
-			description: displayPartsToHtml(property.symbol.getDocumentationComment(this.typeChecker)),
+			...this.visitNamedDeclaration(property),
+			defaultValue: property.initializer?.getText(),
 		};
 	}
 
-	visitType(node) {
+	visitType(node: Node & { type? }) {
```
Comment (jnizet) [inline misc/api-doc.ts:395]: missing type for type?

**Comment by jnizet** [comment]
Also, in my Angular 15 upgrade PR, I commented:

> I had to leave the polyfills.ts file in the e2e-app/src directory. Once the custom-webpack builder supports Angular 15, we should remove this file, remove its reference in e2e-app/tsconfig.json, and replace its reference in angular.json by ["zone.js"]

Since the custom webpack builder now supports Angular 15, I guess we could now do the above (but I haven't tested it)

File: misc/api-doc.ts (hunk 2 above)
Comment (maxokorokov) [reply misc/api-doc.ts:98]: Will check that, thanks

**Pull request merged**